conans/client/conf/__init__.py

```diff
@@ -290,8 +290,8 @@ def default_profile(self):
         ret = os.environ.get("CONAN_DEFAULT_PROFILE_PATH", None)
         if ret:
             if not os.path.isabs(ret):
-                raise ConanException("Environment variable 'CONAN_DEFAULT_PROFILE_PATH' must point "
-                                     "to an absolute path.")
+                ret = os.path.abspath(os.path.join(os.path.dirname(self.filename), ret))
```

Comment:
This does not resolve to the default "profiles" folder.

Reply:
Done, check last commit

Although now I don't like the private import to avoid the circular import error... here I should know only about ``conan.conf``, and now I know things related to ``ClientCache`` 😕

Reply:
But then we don't know if the default_profile comes from the environment variable or not, and we cannot check if it exists consistently (@lasote and I agree that if it comes from the env variable the file is required to exist).

Reply:
Get the env-var in ``ClientCache.default_profile_path``